XWebView/XWVHttpConnection.swift

```diff
@@ -0,0 +1,280 @@
+/*
+ Copyright 2015 XWebView
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+import Foundation
+
+protocol XWVHttpConnectionDelegate {
+    func handleRequest(request: NSURLRequest) -> NSHTTPURLResponse
+    func didOpenConnection(connection: XWVHttpConnection)
+    func didCloseConnection(connection: XWVHttpConnection)
+}
+
+final class XWVHttpConnection : NSObject {
+    private let handle: CFSocketNativeHandle
+    private let delegate: XWVHttpConnectionDelegate
+    private var input: NSInputStream!
+    private var output: NSOutputStream!
+    private let bufferMaxSize = 64 * 1024
+
+    // input state
+    private var requestQueue = [NSURLRequest]()
+    private var inputBuffer: NSMutableData!
+    private var cursor: Int = 0
+
+    // output state
+    private var outputBuffer: NSData!
+    private var bytesRemained: Int = 0
+    private var fileHandle: NSFileHandle!
+    private var fileSize: Int = 0
+
+    init(handle: CFSocketNativeHandle, delegate: XWVHttpConnectionDelegate) {
+        self.handle = handle
+        self.delegate = delegate
+        super.init()
+    }
+
+    func open() -> Bool {
+        let ptr1 = UnsafeMutablePointer<Unmanaged<CFReadStream>?>.alloc(1)
+        let ptr2 = UnsafeMutablePointer<Unmanaged<CFWriteStream>?>.alloc(1)
+        defer {
+            ptr1.dealloc(1)
+            ptr2.dealloc(1)
+        }
+        CFStreamCreatePairWithSocket(nil, handle, ptr1, ptr2)
+        if ptr1.memory == nil || ptr2.memory == nil { return false }
+
+        input = ptr1.memory!.takeRetainedValue()
+        output = ptr2.memory!.takeRetainedValue()
+        CFReadStreamSetProperty(input, kCFStreamPropertyShouldCloseNativeSocket, kCFBooleanTrue)
+        CFWriteStreamSetProperty(output, kCFStreamPropertyShouldCloseNativeSocket, kCFBooleanTrue)
+
+        input.delegate = self
+        output.delegate = self
+        input.scheduleInRunLoop(NSRunLoop.currentRunLoop(), forMode: NSDefaultRunLoopMode)
+        output.scheduleInRunLoop(NSRunLoop.currentRunLoop(), forMode: NSDefaultRunLoopMode)
```

Comment:
I'm not quite understand your meaning. For threading, I don't think it should run on  main thread. The currentLoop is on a dedicated thread for http server. This is same with old objc implementation version.